packages/frontend/src/components/SchedulersView/SchedulersViewUtils.tsx

```diff
@@ -0,0 +1,124 @@
+import { SchedulerJobStatus, SchedulerWithLogs } from '@lightdash/common';
+import { MantineTheme, Tooltip } from '@mantine/core';
+import {
+    IconAlertTriangleFilled,
+    IconCircleCheckFilled,
+    IconClockFilled,
+    IconCsv,
+    IconPhoto,
+    IconProgress,
+} from '@tabler/icons-react';
+import React from 'react';
+import MantineIcon from '../common/MantineIcon';
+import { IconBox } from '../common/ResourceIcon';
+
+export type SchedulerItem = SchedulerWithLogs['schedulers'][number];
+export type Log = SchedulerWithLogs['logs'][number];
+
+type ColumnName =
+    | 'name'
+    | 'destinations'
+    | 'frequency'
+    | 'lastDelivery'
+    | 'actions'
+    | 'jobs'
+    | 'deliveryScheduled'
+    | 'deliveryStarted'
+    | 'status';
+
+export interface Column {
+    id: ColumnName;
+    label?: string;
+    cell: (item: SchedulerItem) => React.ReactNode;
+    meta?: {
+        style: React.CSSProperties;
+    };
+}
+
+export const getSchedulerIcon = (item: SchedulerItem, theme: MantineTheme) => {
+    switch (item.format) {
+        case 'csv':
+            return (
+                <IconBox
+                    icon={IconCsv}
+                    color="indigo.6"
+                    style={{ color: theme.colors.indigo[6] }}
+                />
+            );
+        case 'image':
+            return (
+                <IconBox
+                    icon={IconPhoto}
+                    color="indigo.6"
+                    style={{ color: theme.colors.indigo[6] }}
+                />
+            );
```

Comment:
can you add default case with assertUnreachable check? it's always nice to have exhaustive switch blocks